packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -1490,15 +1490,26 @@ class FocusManager with DiagnosticableTreeMixin, ChangeNotifier {
   ///
   /// If [highlightMode] returns [FocusHighlightMode.traditional], then widgets should
   /// draw their focus highlight whenever they are focused.
-  // Don't want to set _highlightMode here, since it's possible for the target
-  // platform to change (especially in tests).
-  FocusHighlightMode get highlightMode => _highlightMode ?? _defaultModeForPlatform;
-  FocusHighlightMode _highlightMode;
+  FocusHighlightMode get highlightMode => _highlightMode;
+  FocusHighlightMode _highlightMode = _defaultModeForPlatform;
 
   // If set, indicates if the last interaction detected was touch or not.
   // If null, no interactions have occurred yet.
   bool _lastInteractionWasTouch;
 
+  /// Updates initial value of highlight mode.
+  ///
+  /// It necessary when platform is changed during testing
+  /// (Since working with FocusManager is carried out using singleton pattern).
```

Comment:
Yes, the focus manager is a singleton, but it is reset to a new one after every test (and for each target platform test variant), so this shouldn't be necessary.

Here's [where it is reset](https://cs.opensource.google/flutter/flutter/+/master:packages/flutter_test/lib/src/binding.dart;l=861;drc=a5fa083906fcaf88b039a717c6e78b9814f3a77c).